src/main/java/picard/fastq/BamToBfqWriter.java

```diff
@@ -1,444 +1,446 @@
+        int fileIndex = 0;
+        initializeNextBfqFiles(fileIndex++);
+
+        int records = 0;
+
+        final FilteringSamIterator it = new FilteringSamIterator(iterator, new AggregateFilter(filters));
+        while (it.hasNext()) {
+            final SAMRecord record = it.next();
+            records++;
+            if (records % increment == 0) {
+
+                record.setReadName(record.getReadName() + "/1");
+                writeFastqRecord(codec1, record);
+                wrote++;
+                if (wrote % 1000000 == 0) {
+                    log.info(wrote + " records processed.");
+                }
+                if (chunk > 0 && wrote % chunk == 0) {
+                    initializeNextBfqFiles(fileIndex++);
+                }
+            }
+        }
+    }
+
+    /**
+     * Closes any the open bfq file(s), if any, and opens the new one(s)
+     *
+     * @param fileIndex the index (counter) of the files to write
+     */
+    private void initializeNextBfqFiles(final int fileIndex) {
+        // Close the codecs if they were writing before
+        if (codec1 != null) {
+            codec1.close();
+            if (pairedReads) {
+                codec2.close();
+            }
+        }
+
+        // Open new file, using the fileIndex.
+        final File bfq1 = getOutputFile(this.outputPrefix , 1, fileIndex);
+        codec1 = new BinaryCodec(IOUtil.openFileForWriting(bfq1));
+        log.info("Now writing to file " + bfq1.getAbsolutePath());
+        if (pairedReads) {
+            final File bfq2 = getOutputFile(this.outputPrefix , 2, fileIndex);
+            codec2 = new BinaryCodec(IOUtil.openFileForWriting(bfq2));
+            log.info("Now writing to file " + bfq2.getAbsolutePath());
+        }
+    }
+
+    /**
+     * Writes out a SAMRecord in Maq fastq format
+     *
+     * @param codec the code to write to
+     * @param rec   the SAMRecord to write
+     */
+    private void writeFastqRecord(final BinaryCodec codec, final SAMRecord rec) {
+
+        // Trim the run barcode off the read name
+        String readName = rec.getReadName();
+        if (namePrefix != null && readName.startsWith(namePrefix)) {
+            readName = readName.substring(nameTrim);
+        }
+        // Writes the length of the read name and then the name (null-terminated)
+        codec.writeString(readName, true, true);
+
+        final char[] seqs = rec.getReadString().toCharArray();
+        final char[] quals = rec.getBaseQualityString().toCharArray();
+
+        int retainedLength = seqs.length;
+        if (clipAdapters){
+            // adjust to a shorter length iff clipping tag exists
+            Integer trimPoint = rec.getIntegerAttribute(ReservedTagConstants.XT);
+            if (trimPoint != null) {
+                ValidationUtils.validateArg(rec.getReadLength() == seqs.length, () -> "length of read and seqs differ. Found " + rec.getReadLength() + " and '" + seqs.length + ".");
```

Comment:
this is the actual change in the file...everything else is due to newline differences.....